exp/api/v1beta1/ipam_types.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	// IPPoolLabel is used to add the pool name to an IPAddressClaim and IPAddress.
+	IPPoolLabel = "ipam.cluster.x-k8s.io/pool"
+)
+
+// IPAddressSpec is the desired state of an IPAddress.
+type IPAddressSpec struct {
+	// ClaimRef is a reference to the claim this IPAddress was created for.
+	ClaimRef corev1.LocalObjectReference `json:"claim,omitempty"`
+
+	// PoolRef is a reference to the pool that this IPAddress was created from.
+	PoolRef corev1.TypedLocalObjectReference `json:"pool,omitempty"`
+
+	// Address is the IP address.
+	Address string `json:"address"`
+
+	// Prefix is the prefix of the address.
+	Prefix int `json:"prefix"`
+
+	// Gateway is the network gateway of network the address is from.
+	Gateway string `json:"gateway"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=ipaddresses,scope=Namespaced,categories=cluster-api
+// +kubebuilder:subresource:status
```

Comment:
We currently do not. Probably doesn't make sense to have that comment there, even though generation does not complain. I've removed it for now. If we add a status later, we just have to remember to re-add it again.

Reply:
Sounds reasonable. Yup generation does not complain but it will add a status subresource to the CRD which is not really needed as there's no status to update (but no strong opinions from my side, up to you)